node/consensus/state/state.go

```diff
@@ -658,3 +704,13 @@ func shardExists(shard types.ShardID, shardTerms []ShardTerm) (int, bool) {
 	}
 	return -1, false
 }
+
+func ExtractConfigRequest(ces []ControlEvent) *ConfigRequest {
+	// TODO: decide how to handle multiple config requests
+	for _, ce := range ces {
+		if ce.ConfigRequest != nil {
+			return ce.ConfigRequest
+		}
+	}
+	return nil
+}
```

Comment:
In the context here, extract all into a slice. The decision happens by the caller of this function.